packages/data-provider/test/utils/fileUtils.spec.ts

```diff
@@ -0,0 +1,29 @@
+import { expect } from 'chai';
+import fs from 'fs';
+import { describe, it } from 'mocha';
+import sinon from 'sinon';
+
+const fsPromises = fs.promises;
+
+import { tsvToJson } from '../../src/utils/fileUtils.js';
+
+describe('File Utils tests', () => {
```

Comment:
are there any failure cases we should test? behaviour we expect when the file is not a tsv? empty file behaviour? your example only has a single record, should we test one with multiple records? a header row and no records?

Reply:
adding more tests to cover more edge scenarios